frontend/javascripts/test/puppeteer/dataset_rendering_helpers.js

```diff
@@ -124,17 +127,29 @@ async function waitForTracingViewLoad(page: Page) {
 }
 
 async function waitForRenderingFinish(page: Page) {
+  const width = 1920;
+  const height = 1080;
   let currentShot;
   let lastShot = await page.screenshot({ fullPage: true });
   let changedPixels = Infinity;
   // If the screenshot of the page didn't change in the last x seconds, rendering should be finished
-  while (currentShot == null || changedPixels > 0) {
+  while (currentShot == null || !isPixelEquivalent(changedPixels, width, height)) {
+    console.log(`Waiting for rendering to finish. Changed pixels: ${changedPixels}`);
     await page.waitForTimeout(10000);
     currentShot = await page.screenshot({ fullPage: true });
     if (lastShot != null) {
-      changedPixels = pixelmatch(lastShot, currentShot, {}, 1920, 1080, {
-        threshold: 0.0,
-      });
+      changedPixels = pixelmatch(
+        // The buffers need to be converted to png before comparing them
```

Comment:
Has this become necessary due to the update to `pixelmatch`? Or has this always been an issue? Good catch, either way :+1:

Reply:
From what I read and experienced, `pixelmatch` started to assert this in the new version but the problem existed before. Usually the difference was fairly small (which is why it mostly worked before), but I also encountered garbled diffs during my testing with other `--use-gl` flags.